mvcc/kvstore_txn.go

```diff
@@ -46,7 +63,9 @@ func (tr *storeTxnRead) Range(key, end []byte, ro RangeOptions) (r *RangeResult,
 }
 
 func (tr *storeTxnRead) End() {
-	tr.tx.Unlock()
+	if tr.txlocked {
+		tr.tx.Unlock()
```

Comment:
`tr.txlocked = false`?

Reply:
Can this be better encapsulated? Could `tr.tx.Unlock()` and `tr.txlocked = false` be coordinated by a function that `tr` provides so we have to remember both correctly everywhere?  Same for below `tr.tx.Lock()` logic.